### ISSUE TITLE
MultiIndex coordinates loc/sel not working in 0.19.0

### ISSUE THREAD

**Issue by genric** [pydata/xarray#5691]
**What happened**:
```
TypeError: unhashable type: 'numpy.ndarray'
```

**What you expected to happen**:
No error

**Minimal Complete Verifiable Example**:
Take [this](http://xarray.pydata.org/en/stable/user-guide/data-structures.html#multiindex-coordinates) for example:
```python
import numpy as np
import pandas as pd
import xarray as xr
midx = pd.MultiIndex.from_arrays([["R", "R", "V", "V"], [0.1, 0.2, 0.7, 0.9]], names=("band", "wn"))
mda = xr.DataArray(np.random.rand(4), coords={"spec": midx}, dims="spec")
mda.loc[{'band': 'R', 'wn': 0.1}]
```

**Environment**:
<details>
xarray: 0.19.0,
pandas: 1.3.1,
numpy: 1.21.1,
</details>

**Comment by benbovy** [comment]
Thanks for the report @genric. Probably some regression introduced in #5322. I'm surprised this hasn't been caught in the tests. I'm also surprised that it seems to be fixed in #5692. I'll take a closer look.

**Comment by benbovy** [comment]
So the issue was that #3153 doesn't work well with multi-indexes that don't like numpy arrays nor even numpy scalars given as labels. Previously it worked because of multi-index virtual coordinates, i.e., the `dtype` of multi-index levels was ignored when internally re-organizing the indexers passed to `.sel` in xarray. However, this is not the case anymore as we are currently dropping virtual coordinates by real coordinates in the context of xarray's explicit indexes refactoring.

This should be fixed in #5692 (https://github.com/pydata/xarray/pull/5692/commits/a551c7f05abf90a492fb59068b59ebb2bac8cb4c), although #3153 still doesn't address the case of multi-index level coordinates with float32 values.
